src/s3fs.cpp

```diff
@@ -3739,6 +3739,9 @@ static size_t parse_xattrs(const std::string& strxattrs, xattrs_t& xattrs)
             // something format error, so skip this.
             continue;
         }
+        if(xattrs.end() != xattrs.find(key)){
+            xattrs.erase(key);
+        }
         xattrs.emplace(std::move(key), std::move(val));
```

Comment:
Oh, I didn't realize this.  Maybe `xattrs[std::move(key] = std::move(val)` is a better replacement?

Reply:
Yes, I think you can overwrite it with array access.
But in that case the constructor may have been called. (unconfirmed)

Reply:
Right.  I think this is fine for the case of `std::string` due to small string optimization but `insert_or_assign` is the best way to do this in C++17.  But in any case we should prefer readability over efficiency.

Reply:
The cost efficiency of moving this std::string is trivial, so let's go with array access.
I will fix and update.